packages/playground/website/src/lib/router-hooks.ts

```diff
@@ -0,0 +1,107 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { useSearch, useLocation } from 'wouter';
+
+export function useSearchParams() {
+	const search = useSearch();
+	const location = useCurrentUrl();
+	const [, setLocation] = useLocation();
+	return [
+		useMemo(
+			() => new URL('?' + search, window.location.href).searchParams,
+			[search]
+		),
+		useCallback(
+			(params: Record<string, string | undefined>) => {
```

Comment:
What is the value of using `useCallback()` here? Is it to avoid unnecessarily paying for creating a new closure on each render?

Reply:
Good spot! It's probably excessive and could be removed